_topics/en/transaction-pinning.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Transaction pinning
+
+## Optional.  An entry will be added to the topics index for each alias
+#aliases:
+#  - Foo
+
+## Required.  At least one category to which this topic belongs.  See
+## schema for options
+categories:
+  - Security Problems
+  - Fee Management
+  - Contract Protocols
+  - Transaction Relay Policy
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+excerpt: >
+  **Transaction pinning** is a method for making fee bumping
+  prohibitively expensive by abusing node protections against
+  attacks that can waste bandwidth, CPU, and memory.  This can make fee
+  management more difficult in multipart contract protocols (such as LN).
+
+## Optional.  Use Markdown formatting.  Multiple paragraphs.  Links allowed.
+extended_summary: |
+  Nodes such as Bitcoin Core that allow transactions to be replaced
+  (RBF) or packaged with higher-fee child transactions (CPFP) place
+  restrictions on those replacements in order to prevent various DoS
+  attacks.  However, when two or more people each have the ability to
+  fee bump a transaction, this makes it possible for one of them to
+  pin their version of a transaction at one of the limits and prevent
+  other participants from using fee bumping.
+
+  Some of the limits that can be abused to enable transaction pinning
+  include:
+
+  - **[BIP125][] RBF rule #3** requires a replacement transaction
+    pay a higher absolute fee (not just feerate) than the transaction
+    being replaced and any of its children.  This can allow an attacker
+    to attach a large and low-feerate transaction to the transaction
+    they want to pin, forcing any fee bump to pay for pay for the
+    replacement of the large child transaction.  E.g., with the 2019
+    Bitcoin Core defaults, an attacker can require an honest participant
+    pay a minimum of 0.002 BTC to fee bump a transaction (or even
+    greater amounts in some cases).  <!-- 0.002 BTC = 100,000 vbyte tx
+    at minimum relay fee of 0.00001 BTC/kvB plus minimum incremental
+    relay fee also of 0.00001 BTC/kvB -->
+
+  - **Maximum package size limitations** prevent CPFP from being used if
+    a transaction has more than 101,000 vbytes of children or other
+    descendants in a mempool, or has more than 25 descendants or
+    ancestors.  This can allow an attacker to completely block fee
+    bumping by creating the maximum amount of child transactions.  If
+    the attacker has to create those transactions for other reasons
+    (e.g. because they operate a service paying to users), this attack
+    can be free.  For some two-party contract protocols (such as current
+    LN), this may be mitigated by CPFP carve out.
+
+## Optional.  Produces a Markdown link with either "[title][]" or
+## "[title](link)"
+primary_sources:
+    - title: What is meant by transaction pinning?
```

Comment:
I think this should be filed under 'see also' rather than 'primary sources'. I'm sure I'm not the primary source for transaction pinning!

Reply:
edit: I'm not sure if this link is even needed, since you (indirectly) link to it in the optech mentions section.

Reply:
done